src/raft.rs

```diff
@@ -836,6 +872,83 @@ impl<T: Storage> RaftCore<T> {
         true
     }
 
+    // Pack MsgAppend according to forward info, and send it to target peer.
+    fn send_forward(
+        &mut self,
+        from: u64,
+        commit: u64,
+        commit_term: u64,
+        forward: &Forward,
+        msgs: &mut Vec<Message>,
+    ) {
+        // initialize MsgAppend
+        let mut m = Message::default();
+        m.to = forward.to;
+        m.from = from;
+        m.commit = commit;
+        m.commit_term = commit_term;
+        m.set_msg_type(MessageType::MsgAppend);
+
+        // If log_term and index in forward info mismatch with agent's raft log,
+        // the agent just sends empty MsgAppend.
+        // Empty MsgAppend is only to update commit or trigger decrementing next index .
+        if !self
+            .raft_log
+            .match_term(forward.get_index(), forward.get_log_term())
+        {
+            m.index = forward.get_index();
+            m.log_term = forward.get_log_term();
+            self.send(m, msgs);
+            warn!(
+                self.logger,
+                "index {}, log term {} in forward message to peer {}, do not match the agent's raft log.",
+                forward.get_index(),
+                forward.get_log_term(),
+                forward.get_to()
+            );
+            return;
+        }
+
+        // Fetch log entries from index in forward info to the last index of log.
+        let ents = self.raft_log.entries(
+            forward.get_index() + 1,
+            self.max_msg_size,
+            GetEntriesContext(GetEntriesFor::SendForward {
+                from,
+                commit,
+                commit_term,
+                term: self.term,
+                forward: forward.clone(),
+            }),
+        );
+
+        match ents {
+            Ok(ents) => {
+                m.index = forward.get_index();
+                m.log_term = forward.get_log_term();
+                m.set_entries(ents.into());
+                self.send(m, msgs);
+            }
+            // TODO: Consider a better processing for temporary unavailable in async fetch,
```

Comment:
What about not using async fetch in this case? Since it's the follower replication, it's unnecessary to use async fetch which primarily is for reduce IO latency in raftstore thread.